src/traits.jl

```diff
@@ -53,53 +53,53 @@ Specifically, for any `Color{T,3}`,
 returns true.
 """
 comp1(c::AbstractRGB) = red(c)
-comp1{C<:AbstractRGB}(c::Union{AlphaColor{C},ColorAlpha{C}}) = red(c)
+comp1(c::Union{AlphaColor{C},ColorAlpha{C}}) where {C<:AbstractRGB} = red(c)
 comp1(c::Union{Color,ColorAlpha}) = getfield(c, 1)
 comp1(c::AlphaColor) = getfield(c, 2)
 comp1(c::AGray32) = gray(c)
 
 "`comp2(c)` extracts the second constructor argument (see `comp1`)."
 comp2(c::AbstractRGB) = green(c)
-comp2{C<:AbstractRGB}(c::Union{AlphaColor{C},ColorAlpha{C}}) = green(c)
+comp2(c::Union{AlphaColor{C},ColorAlpha{C}}) where {C<:AbstractRGB} = green(c)
 comp2(c::Union{Color,ColorAlpha}) = getfield(c, 2)
 comp2(c::AlphaColor) = getfield(c, 3)
 
 "`comp3(c)` extracts the third constructor argument (see `comp1`)."
 comp3(c::AbstractRGB) = blue(c)
-comp3{C<:AbstractRGB}(c::Union{AlphaColor{C},ColorAlpha{C}}) = blue(c)
+comp3(c::Union{AlphaColor{C},ColorAlpha{C}}) where {C<:AbstractRGB} = blue(c)
 comp3(c::Union{Color,ColorAlpha}) = getfield(c, 3)
 comp3(c::AlphaColor) = getfield(c, 4)
 
 "`color(c)` extracts the opaque color component from a Colorant (e.g., omits the alpha channel, if present)."
 color(c::Color) = c
-color{C,T}(c::TransparentColor{C,T,4}) = C(comp1(c), comp2(c), comp3(c))
-color{C,T}(c::TransparentColor{C,T,2}) = C(comp1(c))
+color(c::TransparentColor{C,T,4}) where {C,T} = C(comp1(c), comp2(c), comp3(c))
+color(c::TransparentColor{C,T,2}) where {C,T} = C(comp1(c))
 
 # Some of these traits exploit a nice trick: for subtypes, walk up the
 # type hierarchy until we get to a stage where we can define the
 # function in general
 
 # recurse up the type hierarchy until you get to Colorant{T,N} for
 # specific T,N.
-to_top{T,N}(::Type{Colorant{T,N}}) = Colorant{T,N}
-@pure to_top{C<:Colorant}(::Type{C}) = to_top(supertype(C))
+to_top(::Type{Colorant{T,N}}) where {T,N} = Colorant{T,N}
+@pure to_top(::Type{C}) where {C<:Colorant} = to_top(supertype(C))
 
 to_top(c::Colorant) = to_top(typeof(c))
 
 # eltype(RGB{Float32}) -> Float32
-eltype{T          }(::Type{Colorant{T}})   = T
-eltype{T,N        }(::Type{Colorant{T,N}}) = T
-@pure eltype{C<:Colorant}(::Type{C}) = eltype(supertype(C))
+eltype(::Type{Colorant{T}}) where {T          }   = T
+eltype(::Type{Colorant{T,N}}) where {T,N        } = T
```

Comment:
Yeah. I wonder if it's something FemtoCleaner could figure out, that it had lined up previously. Dunno though.